### PR TITLE
Add Cluster Deleter

### PR DESCRIPTION
**Pull request by tatlat**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/clustermanager/deleter.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	err := retrier.New(
+		a.deleteClusterTimeout,
+		retrier.WithRetryPolicy(retrier.BackOffPolicy(a.retryBackOff)),
+	).Retry(func() error {
+		var err error
+		client, err = a.clientFactory.BuildClientFromKubeconfig(managementCluster.KubeconfigFile)
+		if err != nil {
+			return errors.Wrap(err, "building client to delete cluster")
+		}
+		spec.Cluster.Namespace = "default"
```
Comment (g-gaston) [inline pkg/clustermanager/deleter.go:89]: I think this is wrong
The user can decide what namespace to use

**Review by g-gaston — COMMENTED** [review]
Very clean, I like it a lot

File: pkg/dependencies/factory.go [hunk 2]
```diff
@@ -1278,6 +1279,27 @@ func (f *Factory) WithClusterApplier() *Factory {
...
 }
 
+// WithClusterDeleter builds a cluster deleter.
+func (f *Factory) WithClusterDeleter() *Factory {
+	f.WithLogger().WithUnAuthKubeClient().WithLogger()
+
+	f.buildSteps = append(f.buildSteps, func(ctx context.Context) error {
+		var opts []clustermanager.DeleterOpt
+		if f.config.noTimeouts {
+			// opts = append(opts, clustermanager.ManagementUpgraderRetrier(*retrier.NewWithNoTimeout()))
```
Comment (mitalipaygude) [inline pkg/dependencies/factory.go:1289]: nit: we can remove this code if it's not needed

**Review by mitalipaygude — APPROVED** [review]

**Review by mitalipaygude — APPROVED** [review]

**Pull request merged**